File: Python/Greedy/programmers_088.py
def solution(people, limit):
    answer = len(people)
    _people = sorted(people, reverse=True)
    start, end = 0, len(people)-1
    while start < end:
        if _people[start]+_people[end] <= limit:
            end -= 1
            answer -= 1
        start += 1

    return answer

File: JavaScript/programmers_088.js
// LV.2 구명보트

function solution(people, limit) {
  let answer = people.length;
  people.sort((a, b) => b - a);
  let start = 0;
  let end = people.length - 1;
  while (start < end) {
    if (people[start] + people[end] <= limit) {
      end -= 1;
      answer -= 1;
    }
    start += 1;
  }
  return answer;
}
